app/src/main/java/db/AppDatabase.kt

```diff
@@ -4,42 +4,78 @@ import android.content.Context
 import androidx.room.Database
 import androidx.room.Room
 import androidx.room.RoomDatabase
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
 import db.dao.CardDAO
 import db.dao.CardStackDAO
-import db.dao.CourseDAO
 import db.entities.Card
 import db.entities.CardStack
-import db.entities.Course
 
-@Database(entities = [Course::class, CardStack::class, Card::class], version = 1)
+@Database(entities = [CardStack::class, Card::class], version = 2)
 abstract class AppDatabase : RoomDatabase() {
-    abstract fun courseDao(): CourseDAO
     abstract fun cardStackDAO(): CardStackDAO
     abstract fun cardDAO(): CardDAO
 
     companion object {
+        const val DB_NAME = "flashcards_db"
         private var instance: AppDatabase? = null
+        val MIGRATION_1_2: Migration = object : Migration(1, 2) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                //CardStack Migration
+                database.execSQL("ALTER TABLE CardStack RENAME TO CardStackOLD")
```

Comment:
you could execute your workaround for CardStack table modification (due to limited sql support from sqlite) as a transaction:

```sql
 database.execSQL("""BEGIN TRANSACTION""")

 database.execSQL("ALTER TABLE CardStack RENAME TO CardStackOLD")
 database.execSQL(
                     "CREATE TABLE CardStack(" +
                             "cardStackId INTEGER NOT NULL PRIMARY KEY," +
                             "name TEXT NOT NULL," +
                             "num_cards INTEGER NOT NULL) "
                 )
                 database.execSQL("INSERT INTO CardStack(cardStackId, name, num_cards) SELECT cardStackId, name, num_cards FROM CardStackOLD")
database.execSQL("DROP TABLE CardStackOLD")

  database.execSQL("""COMMIT""")
```